### DISCUSSION TITLE
Looking for Quality LUA Tuttorials

### DISCUSSION THREAD

**Discussion by FlyingJudgement** [o3de/o3de#11304] [Q&A]
Hi all I'am looking for good Lua tuttorials Paid or unpaid. I Learnt to code in C# This will be my second Language, what is the fastest way to get up to speed?  
Do I need to learn C++ to work in O3DE or is there any benefit?

**Comment by santorac** [comment]
Here is the main page for lua in O3DE. It includes links to some resources for learning lua. https://www.o3de.org/docs/user-guide/scripting/lua/

You don't necessarily need to learn C++ to work with O3DE. There is a lot you can do with lua and/or [Script Canvas](https://www.o3de.org/docs/user-guide/scripting/script-canvas/) to do gameplay scripting. This tutorial might be a good place to start https://www.youtube.com/playlist?list=PLCQwFpnHSZQhhsr6iJqyeYqKe6dexnHh9. You can also customize the editor using python scripts, though I'm not aware of any tutorials for that. There is a new "[Material Canvas](https://github.com/o3de/sig-graphics-audio/issues/51)" tool being developed that will allow users to create custom material behavior using a node graph. If you want to do more advanced things like creating a custom component that will require C++, at least for now.

**Comment by FlyingJudgement** [comment]
Hello Thank you for you reply.
 I picked up C++ already and learned the syntax of lua too(but not
scripting with lua) Im at a high level with C# so C++ was realy easy and a
funn extension. Making complex custome components is what i do daily, Lua
and script cavas isnt for me, I cant see a clean way to use any of them for
a serious maner withouth getting confusing and eventualy unmanagable. Its
great for fast prototyping, artists and moding support. I realy wish lua
and script canvas wouldnt be the focus of O3DE but make C++ fast and easyer
to use withouth it is to limited and slow.
Thank you for your help again, we decided its to risky for us to migrate to
O3DE at this point.
Regards Balint

**Reply by santorac** [reply]
I do believe there are some efforts to make component development easier.... Yeah, I found it, there's a system that uses templates to generate the boilerplate code for the new component.  https://www.o3de.org/docs/user-guide/programming/components/create-component/
